protocol/protocol.go

```diff
@@ -24,19 +24,26 @@ import (
 func (m *Manager) routerInitializer(netConnection *netadapter.NetConnection) *routerpkg.Router {
 	router := routerpkg.NewRouter()
 	spawn("newRouterInitializer-startFlows", func() {
-		if m.context.ConnectionManager().IsBanned(netConnection) {
+		isBanned, err := m.context.ConnectionManager().IsBanned(netConnection)
+		if err != nil {
+			panic(err)
+		}
+		if isBanned {
 			err := m.context.NetAdapter().Disconnect(netConnection)
 			if err != nil {
 				panic(err)
 			}
 			return
 		}
 
-		err := m.startFlows(netConnection, router)
+		err = m.startFlows(netConnection, router)
 		if err != nil {
 			if protocolErr := &(protocolerrors.ProtocolError{}); errors.As(err, &protocolErr) {
 				if protocolErr.ShouldBan {
-					m.context.ConnectionManager().Ban(netConnection)
+					err := m.context.ConnectionManager().Ban(netConnection)
```

Comment:
this shouldn't panic since it's ok for you to not know an address of an inbound connection